js/search/domain.js

```diff
@@ -78,6 +78,10 @@ const Domain = new Lang.Class({
 
         this._load_sync_internal();
         this._shard_inited = true;
+
+        /* Append shards to default EknVfs for ekn:// uri to work */
+        if (this._shards)
+            Eknc.default_vfs_set_shards (this._shards);
```

Comment:
This seems wrong, considering just creating a new domain will overwrite the default shards.